scripts/pyinstaller/make_exe.py

```diff
@@ -89,6 +93,37 @@ def clean_pyinstaller_build_dirs():
         print(f"Deleted build directory: {str(location)}")
 
 
+def zip_with_symlinks(source_dir: Path, output_zip: Path) -> None:
+    """Create a zip archive that preserves symlinks."""
+    output_zip.parent.mkdir(parents=True, exist_ok=True)
+    with zipfile.ZipFile(output_zip, "w", zipfile.ZIP_DEFLATED) as zipf:
+        for root, dirs, files in os.walk(source_dir):
+            # Add directories
+            for dir_name in dirs:
+                dir_path = os.path.join(root, dir_name)
+                if os.path.islink(dir_path):
+                    # Handle symlinked directories
+                    arcname = os.path.relpath(dir_path, source_dir)
+                    info = zipfile.ZipInfo(arcname + "/")
+                    info.create_system = 3  # Unix
+                    info.external_attr = 0o120777 << 16  # symlink file type
```

Comment:
This Python docs do not explain these very well:

*   [`create_system`](https://docs.python.org/3/library/zipfile.html#zipfile.ZipInfo.create_system)

    >   System which created ZIP archive.
*   [`external_attr`](https://docs.python.org/3/library/zipfile.html#zipfile.ZipInfo.external_attr)

    >   External file attributes.

Do you have any other documentation that we can link to that helps document the values here?